Source/Tests/ELFCodeLoader2.h

```diff
@@ -806,6 +809,57 @@ class ELFCodeLoader2 final : public FEXCore::CodeLoader {
     // 0 - MONITOR/MWAIT available in CPL3
     // 1 - FSGSBASE instructions available in CPL3
     HWCap2 = 0;
+
+    // We need to know if we support AVX for AT_MINSIGSTKSZ
+    SupportsAVX = !!(res_1.ecx & (1U << 28));
+  }
+
+  uint64_t CalculateSignalStackSize() const {
+    // We must calculate the required signal stack size that the "kernel" consumes.
+    // For FEX this means the amount of state we store in to the guest stack, not including the amount
+    // that FEX stores in to the host stack as well.
+    //
+    // This needs to match what we do in FEXCore's dispatcher (Which should at some point be moved to the frontend).
+    //
+    // This roughly means that we need to calculate the combined size of:
+    // - xstate or _libc_fstate depending on AVX support
+    // - ucontext_t
+    // - siginfo_t
+    // Size of state requiring to be stored is different between 32-bit and 64-bit.
+
+    uint64_t Result{};
+    if (Is64BitMode()) {
+      Result += sizeof(FEXCore::x86_64::ucontext_t);
+      Result = FEXCore::AlignUp(Result, alignof(FEXCore::x86_64::ucontext_t));
+      if (SupportsAVX) {
+        Result += sizeof(FEXCore::x86_64::xstate);
+        Result = FEXCore::AlignUp(Result, alignof(FEXCore::x86_64::xstate));
+      }
+      else {
+        Result += sizeof(FEXCore::x86_64::_libc_fpstate);
+        Result = FEXCore::AlignUp(Result, alignof(FEXCore::x86_64::_libc_fpstate));
+      }
+
+      Result += sizeof(siginfo_t);
+      Result = FEXCore::AlignUp(Result, alignof(siginfo_t));
+    }
+    else {
+      Result += sizeof(FEXCore::x86::ucontext_t);
+      Result = FEXCore::AlignUp(Result, alignof(FEXCore::x86::ucontext_t));
+      if (SupportsAVX) {
+        Result += sizeof(FEXCore::x86_64::xstate);
```

Comment:
Shouldn't this be `sizeof(FEXCore::x86::xstate)` and the non-avx case be using `sizeof(FEXCore::x86::_libc_fpstate)` instead of the x86_64 versions?

Reply:
Yes. Copy and paste fail, thanks for the catch.